### ISSUE TITLE
"import irods" shouldn't interfere in logging by applications or other modules 

### ISSUE THREAD

**Issue by d-w-moore** [irods/python-irodsclient#660]
This line is very invasive to logging that the application (or other imported modules / libraries) might want to do:
https://github.com/irods/python-irodsclient/blob/5ae9590656b78460b9726ec15cce3ad5f6a243db/irods/__init__.py#L32

`basicConfig` is meant to be called only once within an application, and it probably shouldn't be here.

For one thing, it means you cannot do this and see logging output:
```
import logging
import irods
logging.basicConfig(level = logging.DEBUG)
logging.debug("hello!")
```

**Comment by d-w-moore** [comment]
Once this line is gone from the module, we probably won't see logging output from this as we should:
```
import logging
#with PYTHON_IRODSCLIENT_CONFIGURATION_PATH set and the following line in the config file:
#  data_objects.autoclose: True
import irods   
```
Logging streams should receive the output:
```
WARNING:irods.client_configuration:Invalid configuration format at line 1: 'data_objects.autoclose: True'
```
But they will not, as basicConfig(...) hasn't been called yet.

If this is fine by the application writer, no harm done.  If they desire logging at that point however, we'll need the following modification to the above code:
```
import logging
logging.basicConfig( )
import irods
```

**Comment by d-w-moore** [comment]
On a side note, not sure whether colons should maybe be permitted in the config files.  I can think of no reason why not.  THoughts? If folks are ok with an optional colon being permitted at the end of the dotted name, I'll make another issue.

**Comment by alanking** [comment]
> `basicConfig` is meant to be called only once within an application, and it probably shouldn't be here.

Sounds right to me, yes.

> Once the `basicConfig` call is banished from the module, we probably won't see logging output from this short Python script as we might like to:
> 
> ```
> #!/usr/bin/env python3
> # Run this script with PYTHON_IRODSCLIENT_CONFIGURATION_PATH set and the 
> # following line in the config file:
> #  data_objects.auto_close: True
> import logging
> import irods   
> ```

I think that is okay. Perhaps we can expose a function to validate the configuration (if only for debugging purposes) a la configuration validation in the HTTP and S3 APIs? This could be called at any time and emit a message to the log (or stdout, or whatever) to help with finding issues.

Should we consider using a "lazy" approach to setting the client configuration environment variables? Apologies if this comment is naive - not intimately familiar with the implementation at the moment. :)

> On a side note, not sure whether colons shouldn't maybe be permitted in the config files. I can think of no reason why not. THoughts? If folks are ok with an optional colon being permitted at the end of the dotted name, I'll make another issue.

I don't think we should support colons in this simple configuration syntax. If we want to support syntax like that, we might as well move to using JSON or YAML.

**Comment by alanking** [comment]
Also, this reads like a bug to me, but feel free to change the label.

**Comment by alanking** [comment]
@d-w-moore - Please close if complete. Thanks
